pkg/codegen/schema/loader.go

```diff
@@ -210,3 +250,106 @@ func LoadPackageReference(loader Loader, pkg string, version *semver.Version) (P
 	}
 	return p.Reference(), nil
 }
+
+func (l *pluginLoader) loadSchemaBytes(pkg string, version *semver.Version) ([]byte, *semver.Version, error) {
+	pluginInfo, err := workspace.GetPluginInfo(workspace.ResourcePlugin, pkg, nil, false /* skipMetadata */)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	if version == nil {
+		version = pluginInfo.Version
+	}
+
+	cachedSchemaPath := filepath.Join(pluginInfo.Path, fmt.Sprintf("schema-%v.json", pkg))
```

Comment:
~I am pretty nervous about writing data out to an unverified location. Despite what I mentioned about workspaces earlier, I'd really rather that we only read/write from/to cached schemas that live in the workspace's plugin cache.~

I see that you call this out later--thanks!

Reply:
I'm inclined to agree. It was very handy though to be able to `make install` a provider and have this automatically work, but... I'm not sure we should.

Reply:
I've pushed this logic up into the host, which _may_ compute a `SchemaPath` property and put it on the PluginInfo. The path follows the same convention here, not repeating the version.

I think the only remaining questions here are whether we should put the version in this filename, and if so, should we disallow this optimization when the plugin is found via $PATH to avoid cluttering up other directories with this file. I could go either way on this, for 99.99% of users this cache will improve performance. Is it worth it for local engineering? Probably not.

Reply:
I've removed support for caching locally installed packages in this PR to limit scope.